cookbook/serializer.py

```diff
@@ -45,7 +47,7 @@ def to_internal_value(self, data):
 class SpaceFilterSerializer(serializers.ListSerializer):
 
     def to_representation(self, data):
-        if type(data) == QuerySet and data.query.is_sliced:
+        if (type(data) == QuerySet and data.query.is_sliced) or type(data) == MP_NodeQuerySet:
```

Comment:
can you add a comment here why `type(data) == MP_NodeQuerySet` should not be filtered for space ?

This is an extremely critical security function as not filtering where a filter would be necessary could leak data between spaces so everyone needs to understand why a certain piece of code exists.

Reply:
I'll add a comment - but to discuss it here, the MP_NodeQuerySet doesn't have a filter function and would only come from a API call and not a nested serializer, so matches the same criteria that is_sliced was checking for.

Reply:
I looked into this more - I am not using that mixin for Keywords - it didn't work.  Instead I overrode the queryset in the API for a variety of reasons.   I'll simplify the override to include .filter(space=space) as a final step to ensure that it is always captured.